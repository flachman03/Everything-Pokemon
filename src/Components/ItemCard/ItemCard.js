import React from 'react'
import './ItemCard.scss'

export const ItemCard = ({item}) => {
  return (
    <article className="item-card">
      <img src={item.sprites.default} alt={item.name} className="item-card__image"/>
      <h1 className="item-card__name">{item.name}</h1>
    </article>
  )
}