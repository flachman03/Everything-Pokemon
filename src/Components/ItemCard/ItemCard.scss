.item-card {
  height: 100px;
  width: 150px;
  margin: 10px;
  padding: 0;
  background: hsla(254, 0%, 0%, 0.7);
  border: none;
  color: white;
  font-size: 20px;
  border-radius: 20px;
  outline: none;
  &:hover {
    background: #91D1E2;
  }
  &:active {
    background: red;
  }
}
.item-card__image {
  height: 60px;
  width: 60px;
}

.item-card__name {
  margin: 0;
}