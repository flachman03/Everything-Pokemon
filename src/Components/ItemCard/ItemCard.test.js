import React from 'react'
import { shallow } from 'enzyme'
import { ItemCard } from './ItemCard'

describe('ItemCard', () => {
  let wrapper, mockItem;
  beforeEach(() => {
    mockItem = {name: 'poke ball', sprites: {}}
    wrapper = shallow(<ItemCard item={mockItem}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have an item-card article', () => {
    expect(wrapper.find('.item-card')).toHaveLength(1)
  })

  it('should have an image of item-card__image', () => {
    expect(wrapper.find('.item-card__image')).toHaveLength(1)
  })

  it('should have a header of item-card__name', () => {
    expect(wrapper.find('.item-card__name')).toHaveLength(1)
  })
})