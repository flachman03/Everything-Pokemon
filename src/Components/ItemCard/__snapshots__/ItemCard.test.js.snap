// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard should match the snapshot 1`] = `ShallowWrapper {}`;
